### DISCUSSION TITLE
useInfiniteQuery refetch calls the api with default page number

### DISCUSSION THREAD

**Discussion by Npervic** [TanStack/query#1688] [Q&A]
Hey guys I'm having a problem with re fetching infinite data. I have implemented infinite loading for a table I use in my app to fetch next pages and this works great.

So i call the hook like this:

![Screenshot from 2021-01-25 15-01-42](https://user-images.githubusercontent.com/57952143/105715750-4539b700-5f1e-11eb-8669-bf2ad400ce86.png)

I call this with these settings:

![Screenshot from 2021-01-25 15-06-02](https://user-images.githubusercontent.com/57952143/105716263-e1fc5480-5f1e-11eb-9c1e-b096c24a61c0.png)

Now when calling `refetch` the default page number set in the `queryFn` method (1) is sent for all 3 requests when for example i have 3 pages loaded it calls the back end like this:

![Screenshot from 2021-01-25 15-04-26](https://user-images.githubusercontent.com/57952143/105716471-238cff80-5f1f-11eb-8f43-a22aa8b42b36.png)

When I call get next page I pass the `pageParam` value. This is not the case with refetch so the `pageParam` value defaults to 1.

I'm kinda unsure how refetching works but i figured it should use the `getPreviousPageParam` function that I passed. The API is all ways called the right amount of times but I'm wondering if and how I should maintain what page is current in my state and somehow pass it? Any help is greatly appreciated.

**Comment by TkDodo** [comment]
_could_ be a bug in `refetch` on `infiniteQuery`, not sure. Can you try to reproduce this in a codesandbox?

My other question would be why you would need `refetch` ? There might be better alternatives :)

**Reply by Npervic** [reply]
I'm using re fetch to refresh table data (either on filter change or on refresh button click):

Yeah I struggled to find a real API that's public to mimic table calls but I managed to create an example:
https://codesandbox.io/s/xenodochial-herschel-4vp7o?file=/src/App.js
The example works fine so I'm guessing I messed up in the implementation I have in my real app. What are the alternatives btw?

**Reply by TkDodo** [reply]
> The example works fine so I'm guessing I messed up in the implementation I have in my real app.

i would suggest to now either add more code to the example until it matches your real app, or work backwords from your app until you reach the example. The issue might just be a minor thing :)

> What are the alternatives btw?

if you want a real "refetch" button that refetches whenever you click - the `refetch` method is fine, that's what it's for. If a filter changes and you want to refetch because of that, the declarative approach would be to add the filter value from your local state to the query key, because react-query will automatically refetch when the query key changes. I've written about this in depth here: https://tkdodo.eu/blog/practical-react-query#treat-the-query-key-like-a-dependency-array
